stdlib/src/collections/dict.mojo

```diff
@@ -55,6 +56,44 @@ trait RepresentableKeyElement(KeyElement, Representable):
     pass
 
 
+@always_inline("nodebug")
+fn _hash_str(s: String) -> UInt:
+    """Hash a string using the DJBX33A hash algorithm.
+
+    When the string is small, the default SIMD hash function is not as efficient
+    as the SIMD machinery has some overhead that is not worth it for small data.
+
+    Args:
+        s: The string to hash.
+
+    Returns:
+        A 64-bit hash value. This value is _not_ suitable for cryptographic
+        uses. Its intended usage is for data structures.
+    """
+    var hash = 5381  # typical starting value
+    for c in s.as_string_slice():
+        hash = ((hash << 5) + hash) + ord(c)  # hash * 33 + ord(char)
+    return hash
```

Comment:
Do we need hash by code-points? If not, you can gain free performance by using this:

```suggestion
    hash = 5381  # typical starting value
    for c in s.as_bytes_span():
        hash = hash * 33 + int(c[])  # hash * 33 + ord(char)
    return hash
```

The compiler can turn your `h * 33` into a shift+plus (I checked the generated assembly). LLVM is indeed good at doing this kind of scalar optimisation.